src/main/java/org/cryptomator/windows/keychain/WinHello.java

```diff
@@ -0,0 +1,42 @@
+package org.cryptomator.windows.keychain;
+
+import org.cryptomator.windows.common.NativeLibLoader;
+
+class WinHello {
+
+	/**
+	 * Encrypts the given cleartext using a Windows Hello key.
+	 *
+	 * @param cleartext The cleartext to encrypt.
+	 * @param challenge A challenge, that needs to be provided during {@link #getEncryptionKey(byte[], byte[]) decryption}
+	 * @return The ciphertext or <code>null</code> if encryption failed.
+	 */
+	public byte[] setEncryptionKey(byte[] cleartext, byte[] challenge) {
+		return Native.INSTANCE.setEncryptionKey(cleartext, challenge);
+	}
+
+	/**
+	 * Decrypts the given ciphertext using a Windows Hello key.
+	 *
+	 * @param ciphertext Ciphertext as previously encrypted using {@link #setEncryptionKey(byte[], byte[])}
+	 * @param challenge Same challenge as used in {@link #setEncryptionKey(byte[], byte[])}
+	 * @return The cleartext or <code>null</code> if decryption failed.
+	 */
+	public byte[] getEncryptionKey(byte[] ciphertext, byte[] challenge) {
+		return Native.INSTANCE.getEncryptionKey(ciphertext, challenge);
+	}
+
+	// initialization-on-demand pattern, as loading the .dll is an expensive operation
+	private static class Native {
+		static final Native INSTANCE = new Native();
+
+		private Native() {
+			NativeLibLoader.loadLib();
+		}
+
+		public native byte[] setEncryptionKey(byte[] cleartext, byte[] challenge);
+
+		public native byte[] getEncryptionKey(byte[] ciphertext, byte[] challenge);
+	}
+
```

Comment:
Not necessary. The Native class has private scope and is only accessible from within its enclosing class. Also the native methods are just "delegates", the actual java comments belong to the normal java methods.

Regarding lib loading: We can safely assume, that the required libraries are installed, since Windows is a quite unform OS. 

A getter method is not needed, since the class is declared private and can only be used from within its enclosing class.